scripts/west_commands/runners/uf2.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2023 Peter Johanson <peter@peterjohanson.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''UF2 runner (flash only) for UF2 compatible bootloaders.'''
+
+import os.path
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+try:
+    import psutil  # pylint: disable=unused-import
+    MISSING_PSUTIL = False
+except ImportError:
+    # This can happen when building the documentation for the
+    # runners package if edtlib is not on sys.path. This is fine
+    # to ignore in that case.
+    MISSING_PSUTIL = True
+
+class UF2BinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for bossac.'''
+
+    def __init__(self, cfg, board_id=None):
+        super().__init__(cfg)
+        self.board_id = board_id
+
+    @classmethod
+    def name(cls):
+        return 'uf2'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--board-id', required=False, dest='board_id',
+                            help='Board-ID value to match from INFO_UF2.TXT')
+ 
+    @classmethod
+    def do_create(cls, cfg, args):
+        return UF2BinaryRunner(cfg, board_id=args.board_id)
+
+    def is_uf2_partition(self, part):
+        return os.path.isfile(os.path.join(part.mountpoint, "INFO_UF2.TXT"))
+
+    def get_uf2_info(self, part):
+        lines = open(os.path.join(part.mountpoint, 'INFO_UF2.TXT'), 'r').readlines()
+
+        summary = lines.pop(0)
+
+        return dict((p[0], p[1]) for p in [p.strip().split(': ', 2) for p in lines])
+
+    def match_board_id(self, part):
+        info = self.get_uf2_info(part)
+
+        return info['Board-ID'] == self.board_id
```

Comment:
Should error handling be here to cope if the array does not have this entry?

Reply:
I added handling for this scenario to the `get_uf2_partitions` to warn on the scenario we find UF2 partitions, but end up filtering them all out due to mismatched `Board-ID` value.

Reply:
I think this will throw a `KeyError` if there is no `Board-ID` value in the INFO_UF2.TXT file. You either need to catch that somewhere or use something like `info.get('Board-ID')` which returns a default value instead of throwing.